File: Jan_Basic_Programming/Assignments/C/22.c
#include<stdio.h>
int main(){
    int num, sum = 0;
     printf("Enter the Number : ");
    scanf("%d", &num);

    for(int i=1; i<=num; i++){
        if(num/i == 0){
            sum+=i;
        }
    }

    if(sum == num){
        printf("%d is a perfect Number. ", num);
    }else{
        printf("%d is not a perfect Number. ", num);
    }

 return 0;
}

File: Mar_PHP/Assignments/BasicPrograms/22.php
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Check Perfect Number</title>
</head>
<body>
    <h2>Check Perfect Number</h2>
    <form method="post">
        Enter a number: <input type="number" name="number" required>
        <input type="submit" name="check" value="Check">
    </form>

    <?php
    if (isset($_POST['check'])) {
        $number = $_POST['number'];
        $sum = 0;

        
        for ($i = 1; $i < $number; $i++) {
            if ($number % $i == 0) {
                $sum += $i;
            }
        }

        
        if ($sum == $number) {
            echo "<p>$number is a Perfect Number.</p>";
        } else {
            echo "<p>$number is not a Perfect Number.</p>";
        }
    }
    ?>
</body>
</html>
